numba/np/ufunc/array_exprs.py

```diff
@@ -392,6 +393,10 @@ def generate(self, *args):
             return self.cast(result, inner_sig.return_type,
                              self.outer_sig.return_type)
 
+    # create a fake ufunc object which is enough to trick numpy_ufunc_kernel
+    ufunc = SimpleNamespace(nin=len(expr_args), nout=1, __name__=expr_name)
+    ufunc.nargs = ufunc.nin + ufunc.nout
+
     args = [lowerer.loadvar(name) for name in expr_args]
     return npyimpl.numpy_ufunc_kernel(
-        context, builder, outer_sig, args, ExprKernel, explicit_output=False)
+        context, builder, outer_sig, args, ufunc, ExprKernel)
```

Comment:
Mildly hacky, but easier to reason about than trying to infer `ufunc.nin` and `ufunc.nout` from `sig`, `args`, and `explicit_outputs`.